lib/modules/manager/gomod/extract.spec.ts

```diff
@@ -19,8 +19,17 @@ describe('modules/manager/gomod/extract', () => {
       expect(res?.filter((e) => e.depType === 'replace')).toHaveLength(1);
     });
 
-    it('extracts constraints', () => {
+    it('extracts constraints and golang', () => {
       const res = extractPackageFile(gomod3);
+      expect(res?.deps).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            depType: 'golang',
+            depName: 'go',
+            datasource: 'golang-version',
+          }),
+        ])
+      );
       expect(res).toMatchSnapshot();
```

Comment:
there's no need for this match at all, it's for checking constraints and golang and that's enough imo
it was there since the start and i didn't want to remove it
im removing it now